File: 171.Excel-Sheet-Column-Number.go
// https://leetcode.com/problems/excel-sheet-column-number/
//
// algorithms
// Easy (50.46%)
// Total Accepted:    200,278
// Total Submissions: 396,896

package leetcode

func titleToNumber(s string) int {
	length := len(s)
	if length == 0 {
		return 0
	}

	base, res := 1, 0
	for i := length - 1; i >= 0; i-- {
		res += int(s[i]-'A'+1) * base
		base *= 26
	}

	return res
}

File: 171.Excel-Sheet-Column-Number.java
// https://leetcode.com/problems/excel-sheet-column-number/
//
// algorithms
// Easy (52.25%)
// Total Accepted:    237,025
// Total Submissions: 453,594
// beats 100.0% of java submissions


class Solution {
    public int titleToNumber(String s) {
        char[] chs = s.toCharArray();
        int length = chs.length;

        int base = 1, res = 0;
        for (int i = length - 1; i >= 0; i--) {
            int bit = (chs[i] - 'A') + 1;
            res += bit * base;
            base *= 26;
        }

        return res;
    }
}